src/module/skb/bpf/skb_hook.bpf.c

```diff
@@ -107,6 +119,64 @@ struct skb_data_ref_event {
 	u8 dataref;
 } __attribute__((packed));
 
+/* Must be called with a valid skb pointer */
+static __always_inline int process_skb_arp(struct retis_raw_event *event,
+					   struct skb_config *cfg,
+					   struct sk_buff *skb,
+					   unsigned char *head,
+					   u16 etype, u16 mac, u16 network)
+{
+	struct skb_arp_event *e;
+	struct arphdr *arp;
+	unsigned char *ptr;
+
+	if (!(cfg->sections & BIT(COLLECT_ARP)))
+		return 0;
+
+	arp = (struct arphdr *)(head + network);
+
+#define ARPHRD_ETHER	1
+	/* We only support ARP for IPv4 over Ethernet */
+	if (BPF_CORE_READ(arp, ar_hrd) != bpf_htons(ARPHRD_ETHER) ||
+	    BPF_CORE_READ(arp, ar_pro) != bpf_htons(ETH_P_IP))
+		return 0;
+
+#define ARPOP_REQUEST	1
+#define ARPOP_REPLY	2
+	/* We only support ARP request & reply */
+	if (BPF_CORE_READ(arp, ar_op) != bpf_htons(ARPOP_REQUEST) &&
+	    BPF_CORE_READ(arp, ar_op) != bpf_htons(ARPOP_REPLY))
+		return 0;
+
+	/* h/w addr len must be 6 (MAC) & protocol addr len 4 (IP) */
+	if (BPF_CORE_READ(arp, ar_hln) != 6 || BPF_CORE_READ(arp, ar_pln) != 4)
+		return 0;
+
+	e = get_event_section(event, COLLECTOR_SKB, COLLECT_ARP, sizeof(*e));
+	if (!e)
+		return 0;
+
+	bpf_probe_read_kernel(&e->operation, sizeof(e->operation), &arp->ar_op);
+
+	/* Sender hardware address */
+	ptr = (unsigned char *)(arp + 1);
+	bpf_probe_read_kernel(&e->sha, sizeof(e->sha), ptr);
+
+	/* Sender protocol address */
+	ptr += 6 /* addr len */;
+	bpf_probe_read_kernel(&e->spa, sizeof(e->spa), ptr);
+
+	/* Target hardware address */
+	ptr += 4 /* addr len */;
```

Comment:
nit, I'd put "hw addr len" or "proto addr len" or else it's a bit confusing that we sometimes add 4 and sometimes we add 6.